README.md

```diff
@@ -1,40 +1,110 @@
 
+### Test use case sample
 ```
-QCELL_AUTH_URL = os.environ.get("QCELL_AUTH_URL")
-GRANTTYPE = os.environ.get('GRANTTYPE')
-USERNAME = os.environ.get('USERNAME')
-PASSWORD = os.environ.get('PASSWORD')
-QCELL_URL_PAYMENT = os.environ.get("QCELL_URL_INITIATE")
-QCELL_URL_PROCESS = os.environ.get("QCELL_URL_PROCESS")
+PSP_QMONEY_GRANTTYPE=password
+PSP_QMONEY_USERNAME=14001502
+PSP_QMONEY_PASSWORD=Nhia@123
+PSP_QMONEY_AUTH_URL=https://uat-adpelite.qmoney.gm/login
+PSP_QMONEY_URL_INITIATE=https://uat-adpelite.qmoney.gm/getMoney
+PSP_QMONEY_URL_PROCESS=https://uat-adpelite.qmoney.gm/verifyCode
 ```
 
+
+## Configuration in `openimis-be_py/openIMIS/openIMIS/settings.py`
+
+To integrate Qmoney into your application, add the following code to your `openimis-be_py/openIMIS/openIMIS/settings.py` file:
+
+```python
+PSP_QMONEY_AUTH_URL = os.environ.get("PSP_QMONEY_AUTH_URL")
+PSP_QMONEY_GRANTTYPE = os.environ.get('PSP_QMONEY_GRANTTYPE')
+PSP_QMONEY_USERNAME = os.environ.get('PSP_QMONEY_USERNAME')
+PSP_QMONEY_PASSWORD = os.environ.get('PSP_QMONEY_PASSWORD')
+PSP_QMONEY_URL_PAYMENT = os.environ.get("PSP_QMONEY_URL_INITIATE")
+PSP_QMONEY_URL_PROCESS = os.environ.get("PSP_QMONEY_URL_PROCESS")
+```
+This configuration code is necessary to enable Qmoney integration within your application and should be placed in the specified settings file.
+
+## Creating a Payment Service Provider for Qmoney
+
+To facilitate integration with Qmoney, we've been provided with a test account that allows us to receive money. Below are the test account details:
+
+- `psp_name`: Qmoney
+- `psp_account`: 14001502
+- `psp_pin`: 1234
+
+These details have been included in our migration file. When you run the migration, it will automatically create the Payment Service Provider for Qmoney using the provided test account details. Please note that these details are intended for testing purposes, and if you plan to move to a production environment, you can update them accordingly to reflect the production configuration.
+
+### Add Qmoney Access Token
+
+In order to make requests to any of the Qmoney endpoints, you need an access token. The generation of the access token is already handled in the code, and all you have to do is add the following details in the `Api_utlity` model:
+
+- `name`: Qcell_token
+- `access_token`: `<generated access_token>`
+- `access_TokenExpiry`: `<access_token expiry date, e.g., 2023-08-07 14:58:39.136>`
+
+For Qmoney integration, the initial access token details are added during the migration process. If you are planning to transition to a production environment, you can choose to update the access_token field and the expiry date as needed. It's important to note that if the access token expires, the system automatically sends a new request to update the access token field. For more details on the implementation, you can refer to `mobile_payment/utils.py`.
```

Comment:
so if you want to retrieve initial access_token you can use post man or use the following code import requests
import json
import os
import settings
from datetime import datetime, timedelta
```
auth_url ="https://uat-adpelite.qmoney.gm/login"
data ={  
    "grantType" :"password", 
    "username"  :"14001502", 
    "password"  :"Nhia@123" 
   }


headers = {
           "Content-Type": "application/json", 
           "Authorization": "Basic TkhJQV9BQ0NFU1NfQ0hBTk5FTDpuaGlhQEAxMjM"
         }


response = requests.post(auth_url, headers=headers, data=json.dumps(data))

if response.status_code == 200:
    response_data = response.json()

    print(response_data)
    new_token = response_data ['data']['access_token']
    expiry_seconds = int(response_data['data']['accessTokenExpiry']) - 60
    new_expiry_date = datetime.now() + timedelta(seconds=expiry_seconds)
    print(f' Token = {new_token}')
    print(f' Expiry = {new_expiry_date}')
    ```

Reply:
This is already implemented if the token is expired it will generate another  one, i just mentioned this here if you wish to add a new one before production . the test case once i gave is just a random access_token

Reply:
ok, what's misleading is that you provide an initial token when it can be retrieved. Let's continue the discussion in the `utils.py` file.